TestSolutions/Lombiq.Analyzers.PackageReference/IndirectPackageReference/IndirectPackageReference.csproj

```diff
@@ -0,0 +1,24 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+  </PropertyGroup>
+
+  <!-- This project demonstrates that the package reference doesn't transfer the analyzer effect to its dependents. If
+       you set ImportPackageAgain to "true" and run static code analysis on this soltuion you will get twice as many
+       warnings. This is the desired behavior, shows that the analyzers only activate when they are explicitly imported.
+  -->
+
+  <ItemGroup Condition="'$(ImportPackageAgain)' == 'true'">
+    <PackageReference Include="Lombiq.Analyzers" Version="2.1.1-osoe-110.14" />
+  </ItemGroup>
+  
+  <!-- Dummy target for when ImportPackageAgain is disabled. -->
+  <Target Name="LombiqNetAnalyzers" Condition="'$(ImportPackageAgain)' != 'true'">
+  </Target>
+
+  <Target Name="CopyAnalyzerViolations" DependsOnTargets="LombiqNetAnalyzers" AfterTargets="LombiqNetAnalyzers">
```

Comment:
From all I understand it seems to me that `DependsOnTargets="LombiqNetAnalyzers"` can be removed everywhere. Having that means that when building `CopyAnalyzerViolations`, MSBuild should first build `LombiqNetAnalyzers`, but you never invoke `CopyAnalyzerViolations` itself directly. Instead, you invoke `LombiqNetAnalyzers` from the build script which then triggers the build of `CopyAnalyzerViolations` thanks to the `AfterTargets="LombiqNetAnalyzers"` property.